lib/spack/spack/modules/common.py

```diff
@@ -741,6 +737,22 @@ def environment_modifications(self):
             exclude=spack.util.environment.is_system_path
         )
 
+        # BlueBrain: we do not generate modules for dependencies, include
+        # their modifications and bin/ directories here
+        #
+        # This should be done here to let the current package prepend paths
+        # properly.
+        #
+        # First do some setup for all dependencies, then execute
+        # modifications
+        for dep in set(spec.traverse(root=False, deptype='run')):
+            dep.package.setup_dependent_package(spec.package.module, spec)
+        for dep in set(spec.traverse(root=False, deptype='run')):
+            if not (dep.external and dep.external_modules):
+                dep.package.setup_run_environment(env)
+                if os.path.isdir(dep.prefix.bin):
+                    env.prepend_path('PATH', dep.prefix.bin)
+
```

Comment:
@matz-e: for my own understanding: is this the difference/addition in our fork compared to upstream? And, are there any other bigger differences between upstream and ours?

The reason I am asking is that in a production deployment using upstream spack, not all modules are enabled anyway. So I assume any other deployments will have similar issues as we had (i.e. need to rely on creating modules for "those packages" and rely on autoload).

Reply:
>The reason I am asking is that in a production deployment using upstream spack, not all modules are enabled anyway.

Upstream by default has everything whitelisted? As soon as some modules are blacklisted, one should have the same issues we observed.

This is basically folding all modules into one, to reproduce `upstream` behavior of loading, e.g., `neuron-0.8.0-deadbeef` as a modules, along with `hdf5-1.12.0-feedbeef` etc, but with a single module rather than a full chain.

Reply:
> Upstream by default has everything whitelisted? As soon as some modules are blacklisted, one should have the same issues we observed.

True, that's the default config. I was just referring to the production deployments on actual systems where people change this behavior.

> This is basically folding all modules into one, to reproduce upstream behavior of loading, e.g., neuron-0.8.0-deadbeef as modules, along with hdf5-1.12.0-feedbeef etc, but with a single module rather than a full chain.

Understood.